arraycontext/container/traversal.py

```diff
@@ -59,11 +62,12 @@
         serialize_container, deserialize_container)
 
 
-# {{{ array container traversal
+# {{{ array container traversal helpers
 
 def _map_array_container_impl(
         f: Callable[[Any], Any],
         ary: ArrayContainerT, *,
+        reduce_func: Callable[[Any, Iterable[Tuple[Any, Any]]], Any] = None,
```

Comment:
Honestly, not sure I'm loving this. (Writing this after reading much of the rest of this PR.) Why not make something specific to reductions instead of trying to generalize this by force? (via lots of inefficient function wrapping)

Reply:
@inducer The main reason I did it like this was because `_multimap_array_container_impl` has a lot more logic in there that I felt squimish about largely copy pasting for the reduction case. I can add an if in there to avoid the wrapping. What do you think?

Reply:
I see your point. How about this: Duplicate `map`, but leave the wild wrap-and-dispatch thing in place for `multimap`? I can't really think of a "pretty" way to deal with this.

Reply:
Done in 94e7c6f.